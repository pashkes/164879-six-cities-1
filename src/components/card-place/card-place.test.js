import React from 'react';
import CardPlace from './card-place.jsx';
import renderer from 'react-test-renderer';

describe(`renders correctly`, () => {
  it(`in the default state`, () => {
    const card = renderer
      .create(<CardPlace
        id={`dsdsd`}
        title={``}
        photoURL={``}
        isPremium={false}
        isFavorite={false}
        price={0}
        type={``}
        rating={0}
      />)
      .toJSON();
    expect(card).toMatchSnapshot();
  });

  it(`if the card is favorite`, () => {
    const cardIsFavorite = renderer
      .create(<CardPlace
        id={`fcdx34`}
        title={``}
        photoURL={``}
        isPremium={false}
        isFavorite={true}
        price={0}
        type={``}
        rating={0}
      />)
      .toJSON();
    expect(cardIsFavorite).toMatchSnapshot();
  });

  it(`if the card is premium`, () => {
    const cardIsPremium = renderer
      .create(<CardPlace
        id={`ds43d`}
        title={``}
        photoURL={``}
        isPremium={true}
        isFavorite={false}
        price={0}
        type={``}
        rating={0}
      />)
      .toJSON();
    expect(cardIsPremium).toMatchSnapshot();
  });
});
