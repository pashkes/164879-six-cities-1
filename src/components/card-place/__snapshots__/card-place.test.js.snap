// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly if the card is favorite 1`] = `
<article
  className=" place-card"
>
  <div
    className="__image-wrapper place-card__image-wrapper"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <img
        alt=""
        className="place-card__image"
        height="200"
        src=""
        width="260"
      />
    </button>
  </div>
  <div
    className="__card-info place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          0
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <withRouter(Connect(FavoriteButton))
        id={414}
        isFavorite={true}
        prefixClass="place-card"
      />
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "0%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <Link
        to="/offer/414"
      />
    </h2>
    <p
      className="place-card__type"
    />
  </div>
</article>
`;

exports[`renders correctly if the card is premium 1`] = `
<article
  className=" place-card"
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="__image-wrapper place-card__image-wrapper"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <img
        alt=""
        className="place-card__image"
        height="200"
        src=""
        width="260"
      />
    </button>
  </div>
  <div
    className="__card-info place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          0
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <withRouter(Connect(FavoriteButton))
        id={454554}
        isFavorite={false}
        prefixClass="place-card"
      />
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "0%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <Link
        to="/offer/454554"
      />
    </h2>
    <p
      className="place-card__type"
    />
  </div>
</article>
`;

exports[`renders correctly in the default state 1`] = `
<article
  className=" place-card"
>
  <div
    className="__image-wrapper place-card__image-wrapper"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <img
        alt=""
        className="place-card__image"
        height="200"
        src=""
        width="260"
      />
    </button>
  </div>
  <div
    className="__card-info place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          0
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <withRouter(Connect(FavoriteButton))
        id={525}
        isFavorite={false}
        prefixClass="place-card"
      />
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "0%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <Link
        to="/offer/525"
      />
    </h2>
    <p
      className="place-card__type"
    />
  </div>
</article>
`;
